addons/mail/models/mail_template.py

```diff
@@ -100,6 +98,26 @@ def _get_non_abstract_models_domain(self):
                                help='The current user can edit the template.')
     is_template_editor = fields.Boolean(compute="_compute_is_template_editor")
 
+    # view display
+    has_dynamic_reports = fields.Boolean(compute='_compute_has_dynamic_reports')
+    has_mail_server = fields.Boolean(compute='_compute_has_mail_server')
+
+    @api.depends('model_id')
+    def _compute_has_dynamic_reports(self):
+        number_of_dynamic_reports_per_model = dict(
+            self.env['ir.actions.report']._read_group(
+                domain=[('binding_model_id', 'in', self.mapped('model_id').ids)],
+                groupby=['binding_model_id'],
+                aggregates=['id:count'],
+                having=[('__count', '>', 0)]))
```

Comment:
```suggestion
        res = self.env['ir.actions.report']._read_group(
            domain=[('binding_model_id', 'in', self.mapped('model_id').ids)],
            groupby=['binding_model_id'],
            aggregates=['id:count'],
            having=[('__count', '>', 0)])
        number_of_dynamic_reports_per_model = {model: count for model, count in res}
```
has the same number of lines/complexity and is maybe more explicit on the `dict` format since you don't have to infer from the _read_group call ?